common/common.scss

```
@@ -1,29 +1,38 @@
+// We only want to override these CSS vars if the setting is configured
+// otherwise we end up with blank values that don't fall back correctly
+
+@mixin isConfigured($property, $value) {
+  @if $value and $value != "" {
+    #{$property}: #{$value};
+  }
+}
+
 :root {
-  --font-family: #{$body_font};
-  --heading-font-family: #{$headline_font};
-  --base-font-size-smaller: #{$smaller-font-size};
-  --base-font-size: #{$normal-font-size};
-  --base-font-size-larger: #{$larger-font-size};
-  --base-font-size-largest: #{$largest-font-size};
+  @include isConfigured("--font-family", $body_font);
```

```suggestion
  --theme-font-family: $body_font;
  --font-family: var(--theme-font-family, initial);
```
Does this work as a css-only solution?

(would be great if we can avoid increasing our dependence on sass)

---

Hmm I see.

Another option: how about setting the default of the theme settings to `"initial"`. Then we'll get our desired behaviour even with a simple:

```
--font-family: #{$body_font};
```